File: Projection/demo/src/main/java/com/kuanquan/demo/MediaInfo.java
package com.kuanquan.demo;

import android.os.Parcel;
import android.os.Parcelable;

import com.kuanquan.demo.enum_p.MediaType;

/**
 * Created by huzongyao on 2018/6/29.
 */

public class MediaInfo implements Parcelable {

    public MediaType mediaType; // 投屏过来的类型 一般有视频、音频、图片等等，这里主要是视屏
    public String title;  // 视屏标题
    public String url;    // 视屏播放链接
    public String albumArtURI; // 专辑图片链接

    public MediaInfo() {
    }

    protected MediaInfo(Parcel in) {
        title = in.readString();
        url = in.readString();
        albumArtURI = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(url);
        dest.writeString(albumArtURI);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<MediaInfo> CREATOR = new Creator<MediaInfo>() {
        @Override
        public MediaInfo createFromParcel(Parcel in) {
            return new MediaInfo(in);
        }

        @Override
        public MediaInfo[] newArray(int size) {
            return new MediaInfo[size];
        }
    };

    @Override
    public String toString() {
        return "MediaInfo{" +
                "mediaType=" + mediaType +
                ", title='" + title + '\'' +
                ", url='" + url + '\'' +
                ", albumArtURI='" + albumArtURI + '\'' +
                '}';
    }
}

File: Projection/test/src/main/java/com/kuanquan/test/projection/MediaInfo.kt
package com.kuanquan.test.projection

import com.kuanquan.test.enum_p.MediaType
import java.io.Serializable

data class MediaInfo(
    /**
     * 投屏过来的类型 一般有视频、音频、图片等等，这里主要是视屏
     */
    var mediaType: MediaType? = null,
    /**
     * 视屏标题
     */
    var title: String? = null,
    /**
     * 视屏播放链接
     */
    var url: String? = null,
    /**
     * 专辑图片链接
     */
    var albumArtURI: String? = null
) : Serializable {

    override fun toString(): String {
        return "MediaInfo(mediaType=$mediaType, title=$title, url=$url, albumArtURI=$albumArtURI)"
    }
}